### PR TITLE
nmstate: add the docstring and comments

### PR DESCRIPTION
**Pull request by karthiksundaravel**
Add the docstrings and comments to the nmstate provider implementation.

### CODE REVIEW

**Review by rjarry — DISMISSED** [review]

File: os_net_config/impl_nmstate.py [hunk 1]
```diff
@@ -273,12 +316,17 @@ class NmstateNetConfig(os_net_config.NetConfig):
...
         self.interface_data = {}
+        # Dict of the vlan data, with keys being the vlan name
         self.vlan_data = {}
+        # Dict of the route data, with keys being the device name
         self.route_data = {}
+        # List of the rules data
         self.rules_data = []
         self.dns_data = {'server': [], 'domain': []}
         self.bridge_data = {}
+        # Dict of the route data, with keys being the device name
```
Comment (vcandapp) [inline os_net_config/impl_nmstate.py:329]: Is this a typo ? Dict of linux_bond ?

File: os_net_config/impl_nmstate.py [hunk 2]
```diff
@@ -337,9 +401,12 @@ def __dump_key_config(self, config, msg="Applying config"):
         logger.info(f"{msg}\n{cfg_dump}")
 
     def get_vf_config(self, sriov_vf):
-        """Identify the nmstate schema for the given VF
+        """Create the nmstate schema for the given VF
```
Comment (vcandapp) [inline os_net_config/impl_nmstate.py:404]: Fn. definition says "get" - but is it actually "set" ?

File: os_net_config/impl_nmstate.py [hunk 3]
```diff
@@ -767,11 +847,19 @@ def generate_rules(self):
...
+        """
         iface_data = self.iface_state(iface)
         if iface_data and Interface.MAC in iface_data:
             return iface_data[Interface.MAC]
 
     def get_ovs_ports(self, members):
+        """Get the ovs ports in nmstate schema format
+
+        :param members: The members of the ovs/ovs-user bridges
+        """
```
Comment (vcandapp) [inline os_net_config/impl_nmstate.py:862]: :returns ?

File: os_net_config/impl_nmstate.py [hunk 4]
```diff
@@ -1489,6 +1646,13 @@ def parse_ovs_extra(self, ovs_extras, name, data):
                 self._ovs_extra_cfg_val(ovs_extra_cmd, cmd_map, data)
 
     def parse_ovs_extra_for_ports(self, ovs_extras, bridge_name, data):
+        """Parse ovs extra for VLAN
+
+        :param ovs_extras: given ovs extra as string
+        :param name: bridge name
```
Comment (vcandapp) [inline os_net_config/impl_nmstate.py:1652]: name - is this VLAN name ?

File: os_net_config/impl_nmstate.py [hunk 5]
```diff
@@ -170,6 +186,13 @@ def is_dict_subset(superset, subset):
...
 
 def _add_sub_tree(data, subtree):
+    """Create a nested dict with the keys mentioned in subtree
+
+    :param data: Starting point for the creation of nested dict
+    :param subtree: list of keys used in the creation of nested
+        dict.
+    :returns: The tip of the nested dict created
+    """
     if data is None:
```
Comment (vcandapp) [inline os_net_config/impl_nmstate.py:196]: Should we mention the exceptions raised like "raises: ConfigErr" ?

Comment (karthiksundaravel) [reply os_net_config/impl_nmstate.py:197]: Done

File: os_net_config/impl_nmstate.py (hunk 3 above)
Comment (karthiksundaravel) [reply os_net_config/impl_nmstate.py:871]: Added

File: os_net_config/impl_nmstate.py (hunk 2 above)
Comment (karthiksundaravel) [reply os_net_config/impl_nmstate.py:406]: Yes, may be in another PR, we could fix it

File: os_net_config/impl_nmstate.py (hunk 4 above)
Comment (karthiksundaravel) [reply os_net_config/impl_nmstate.py]: Its bridge name

File: os_net_config/impl_nmstate.py (hunk 1 above)
Comment (karthiksundaravel) [reply os_net_config/impl_nmstate.py]: Fixed

File: os_net_config/impl_nmstate.py [hunk 6]
```diff
@@ -222,6 +259,13 @@ def set_linux_bonding_options(bond_options, primary_iface=None):
 
 
 def set_ovs_bonding_options(bond_options):
+    """Parse the ovs bond options from templates
```
[the remark this answers is not in the document]
Comment (karthiksundaravel) [reply os_net_config/impl_nmstate.py:263]: Its indeed parsing the bond options and storing the result in nmstate schema format

File: os_net_config/impl_nmstate.py [hunk 7]
```diff
@@ -192,6 +221,14 @@ def parse_bonding_options(bond_options_str):
 
 
 def set_linux_bonding_options(bond_options, primary_iface=None):
+    """Parse the linux bond options from templates
```
[the remark this answers is not in the document]
Comment (karthiksundaravel) [reply os_net_config/impl_nmstate.py:225]: Its indeed parsing the bond options and storing the result in nmstate schema format

**Review by vcandapp — APPROVED** [review]

**Comment by vcandapp** [comment]
lgtm

**Pull request merged**